internal/locate/region_request.go

```diff
@@ -873,15 +873,14 @@ func (s *sendReqState) next(
 		defer s.releaseStoreToken(s.vars.rpcCtx.Store)
 	}
 
-	sendCtx := s.send(bo, req, timeout)
+	canceled := s.send(bo, req, timeout)
 	s.vars.sendTimes++
 
 	if s.vars.err != nil {
 		// Because in rpc logic, context.Cancel() will be transferred to rpcContext.Cancel error. For rpcContext cancel,
 		// we need to retry the request. But for context cancel active, for example, limitExec gets the required rows,
 		// we shouldn't retry the request, it will go to backoff and hang in retry logic.
-		if sendCtx.Err() != nil && errors.Cause(sendCtx.Err()) == context.Canceled {
-			metrics.TiKVRPCErrorCounter.WithLabelValues("context-canceled", storeIDLabel(s.vars.rpcCtx)).Inc()
+		if canceled {
```

Comment:
Isn't the logic equivalent checking `sendCtx` here? :eyes:

Reply:
No, `sendCtx` must be canceled after `send` when using `RPCCanceller` [here](https://github.com/tikv/client-go/blob/5ac118bbc6a1496229704b4954ae30f246247625/internal/locate/region_request.go#L921). It's a mistake made by me in #1565 .